src/cider/exec/nextgen/function/CiderDateFunctions.cpp

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "function/datetime/DateAdd.h"
+
+namespace {
+
+struct OffsetUnit {
+  unsigned dom;  // day-of-month (0-based)
+  unsigned sod;  // second-of-day
+  unsigned doe;  // day-of-era
```

Comment:
what does era mean?

Reply:
kEpochAdjustedDays is days from 01.01.1970 to 01.03.2000. so era is a unit divided by 400 years, and the 0th era starts from 01.03.2000.

https://github.com/intel/BDTK/blob/1a167f0867cc5d9db0d66aefa4f4d545a79eb614/src/cider/exec/nextgen/function/CiderDateFunctions.cpp#L73-L76